go/tasks/pluginmachinery/flytek8s/pod_helper_test.go

```diff
@@ -1013,6 +1014,144 @@ func TestDeterminePrimaryContainerPhase(t *testing.T) {
+		TerminationMessagePath: "/dev/primary-termination-log",
+	}
+
+	podTemplateSpec := v1.PodSpec{
+		Containers: []v1.Container{
+			defaultContainerTemplate,
+			primaryContainerTemplate,
+		},
+		HostNetwork: true,
+		NodeSelector: map[string]string{
+			"foo": "bar",
+		},
+		SchedulerName: "defaultScheduler",
+		Tolerations: []v1.Toleration{
+			v1.Toleration{
+				Key: "foo",
+			},
+		},
+	}
+
+	mergedPodSpec, err := MergePodSpecs(&podTemplateSpec, &podSpec, "foo")
+	assert.Nil(t, err)
+
+	// validate a PodTemplate-only field
+	assert.Equal(t, podTemplateSpec.HostNetwork, mergedPodSpec.HostNetwork)
+	// validate a PodSpec-only field
+	assert.Equal(t, podSpec.Priority, mergedPodSpec.Priority)
+	// validate an overwritten PodTemplate field
+	assert.Equal(t, podSpec.SchedulerName, mergedPodSpec.SchedulerName)
+	// validate a merged map
+	assert.Equal(t, len(podTemplateSpec.NodeSelector)+len(podSpec.NodeSelector), len(mergedPodSpec.NodeSelector))
+	// validate an appended array
+	assert.Equal(t, len(podTemplateSpec.Tolerations)+len(podSpec.Tolerations), len(mergedPodSpec.Tolerations))
+
+	// validate primary container
+	primaryContainer := mergedPodSpec.Containers[0]
+	assert.Equal(t, podSpec.Containers[0].Name, primaryContainer.Name)
+	assert.Equal(t, primaryContainerTemplate.TerminationMessagePath, primaryContainer.TerminationMessagePath)
+
+	// validate default container
+	defaultContainer := mergedPodSpec.Containers[1]
+	assert.Equal(t, podSpec.Containers[1].Name, defaultContainer.Name)
+	assert.Equal(t, defaultContainerTemplate.TerminationMessagePath, defaultContainer.TerminationMessagePath)
+
+}
+
+func TestBuildPodWithSpec2(t *testing.T) {
+	podSpec := v1.PodSpec{
+		Containers: []v1.Container{
+			v1.Container{
+				Name: "foo",
+			},
+			v1.Container{
+				Name: "bar",
+			},
+		},
+	}
+
+	pod, err := BuildPodWithSpec(nil, &podSpec, "foo")
+	assert.Nil(t, err)
+	assert.True(t, reflect.DeepEqual(pod.Spec, podSpec))
+
+	primaryContainerTemplate := v1.Container{
+		Name:                   primaryContainerTemplateName,
+		TerminationMessagePath: "/dev/primary-termination-log",
+	}
+
+	podTemplate := v1.PodTemplate{
+		Template: v1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"fooKey": "barVal",
+				},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					primaryContainerTemplate,
+				},
+			},
+		},
+	}
+
+	pod, err = BuildPodWithSpec(&podTemplate, &podSpec, "foo")
+	assert.Nil(t, err)
+
+	// Test that template podSpec is merged
+	primaryContainer := pod.Spec.Containers[0]
+	assert.Equal(t, podSpec.Containers[0].Name, primaryContainer.Name)
+	assert.Equal(t, primaryContainerTemplate.TerminationMessagePath, primaryContainer.TerminationMessagePath)
+
+	// Test that template object metadata is copied
+	assert.Contains(t, pod.ObjectMeta.Labels, "fooKey")
+	assert.Equal(t, pod.ObjectMeta.Labels["fooKey"], "barVal")
+}
+
 func TestBuildPodWithSpec(t *testing.T) {
```

Comment:
@hamersaw Shell we keep this test (which I currently left in to demonstrate that the changes don't brick the existing tests) despite its logic having been refactored into `TestMergePodSpecs` and `TestBuildPodWithSpec2`?